modules/new_serverpod_auth/serverpod_auth_session/serverpod_auth_session_server/lib/src/business/user_sessions.dart

```diff
@@ -0,0 +1,186 @@
+import 'dart:io';
+
+import 'package:serverpod/protocol.dart';
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_auth_session_server/src/generated/protocol.dart';
+import 'package:serverpod_auth_session_server/src/util/random_string.dart';
+import 'package:serverpod_auth_session_server/src/util/salted_hash.dart';
+import 'package:serverpod_auth_user_server/serverpod_auth_user_server.dart';
+
+/// Management functions for [UserSession]s.
+///
+/// This should be used instead of [UserSession.db].
+abstract final class UserSessions {
+  static const _signInSalt =
+      "session.passwords['authKeySalt'] ?? defaultAuthKeySalt"; // TODO ?
+
+  /// Looks up the `AuthenticationInfo` belonging to the [key].
+  ///
+  /// The [key] maps to the [UserSession.sessionKeyHash] field.
+  ///
+  /// In case the authentication could not be resolved, this returns `null` and does never throw.
+  static Future<AuthenticationInfo?> authenticationHandler(
+    final Session session,
+    final String key,
+  ) async {
+    try {
+      final sessionKeyHash = hashString(_signInSalt, key);
+
+      final tempSession = await session.serverpod.createSession(
```

Comment:
TODO: I need to understand why this is even used. Is that still relevant in the latest version of Serverpod?

As shown below, this does not work in combination with `withServerpod` integration tests for me. In those cases the `tempSession` could not read the data written with the "test session" (maybe because it runs in an open transaction?).

Reply:
Removed the temp session.

We'll need to see whether there is any issue with using `Session` in the authentication handler in the E2E tests later, but for now not using the `tempSession` is the only way to get the integration tests (`withServerpod`) working.